charts/metrics-server/templates/deployment.yaml

```
@@ -33,7 +33,9 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
     spec:
-      schedulerName: {{ .Values.schedulerName }}
+      {{- if .Values.schedulerName }}
+      schedulerName: {{ .Values.schedulerName | quote }}
+      {{- end }}
```

```suggestion
      {{- with .Values.schedulerName }}
      schedulerName: {{ . }}
      {{- end }}
```
This pattern would be more idiomatic for the chart. How come you added the `quote` operator, I can't see a reason why it'd be needed?

---

Hey @stevehipwell, thanks for the feedback !

Indeed after further readings, empty values seems to be valid in YAML 1.2 spec (see [#72-empty-nodes](https://yaml.org/spec/1.2.2/#72-empty-nodes)).

But still we both agree it is a bit unnecessary to have this value set with empty value.

Regarding `with` / `if` your suggestion seems reasonable. I tend to use `with` for mapping value more often that's why I did it this way.

`quote` is just what I believed was best practice, as we know this field must be a string value. Helm can perform some sketchy type juggling from times to times, that's why some other string values are also explicitly quoted in the file, such as [deployment.yaml#L47](https://github.com/kubernetes-sigs/metrics-server/blob/master/charts/metrics-server/templates/deployment.yaml#L47).

I'll update the changelog then !